src/auth/oidc/models/openid.class.spec.ts

```diff
@@ -192,3 +192,93 @@ test('OIDC discover', () => {
     const result = oidc.discover()
     expect(spy).toBeCalled()
 })
+
+test('OIDC authenticate when not authenticated', async () => {
+    const mockRequest = {
+        body: {},
+    } as Request
+    mockRequest.isUnauthenticated = () => {
+        return true
+    }
+    const mockResponse = {} as Response
+    const mockRedirect = jest.fn()
+    mockResponse.redirect = mockRedirect
+
+    const next = jest.fn()
+    await oidc.authenticate(mockRequest, mockResponse, next)
+    expect(mockRedirect).toBeCalledWith(AUTH.ROUTE.LOGIN)
+})
+
+test('OIDC authenticate when authenticated but session and client not initialised', async () => {
+    const mockRequest = {
+        body: {},
+    } as Request
+    mockRequest.isUnauthenticated = () => {
+        return false
+    }
+    const mockResponse = {} as Response
+    const mockRedirect = jest.fn()
+    mockResponse.redirect = mockRedirect
+
+    const next = jest.fn()
+    await oidc.authenticate(mockRequest, mockResponse, next)
+    expect(mockRedirect).toBeCalledWith(AUTH.ROUTE.LOGIN)
+})
+
+test('OIDC initialiseStrategy', async () => {
+    const issuer = {}
+    const spyGetOptions = jest.spyOn(oidc, 'getOpenIDOptions')
+    const spyGetNewStrategy = jest
+        .spyOn(oidc, 'createNewStrategy')
+        .mockImplementation(() => Promise.resolve({} as Strategy<any, any>))
+    const spyUseStrategy = jest.spyOn(oidc, 'useStrategy')
+
+    const options = {
+        authorizationURL: '',
+        tokenURL: '',
+        clientID: 'clientId',
+        clientSecret: 'Clientsecret',
+        discoveryEndpoint: 'someEndpoint',
+        issuerURL: 'issuer_url',
+        logoutURL: 'logouturl',
+        callbackURL: 'redirect_uri',
+        responseTypes: ['none'],
+        scope: 'some scope',
+        sessionKey: 'key',
+        tokenEndpointAuthMethod: 'client_secret_basic',
+        useRoutes: false,
+    }
+
+    await oidc.initialiseStrategy(options)
+    expect(spyGetOptions).toBeCalled()
+    expect(spyGetNewStrategy).toBeCalled()
+    expect(spyUseStrategy).toBeCalled()
+})
+
+test('test createNewStrategy', async () => {
+    /* eslint-disable @typescript-eslint/camelcase */
+    const options = {
```

Comment:
If you wanted to trim this down, you *might* be able to use createMock<OpenIDMetadata>()

Reply:
I have to stub here. Validation might fail for redirect_url